disnake/interactions/modal.py

```diff
@@ -122,4 +128,4 @@ class ModalInteractionData:
 
     def __init__(self, *, data: ModalInteractionDataPayload):
         self.custom_id: str = data["custom_id"]
-        self.components: List[ActionRowPayload] = data["components"]
+        self.components: List[ActionRow] = [ActionRow(d) for d in data["components"]]
```

Comment:
I like this idea, very good, but I have a suggestion.

Sadly, because of the annotation of `components.ActionRow.children`, you have a bad annotation which isn't accurate, my suggestion is:

1. Add a `from_component` classmethod to `ui.InputText` (like every other component).
2. Call `components._component_factory` and send that to the classmethod.
3. Use `ui.ActionRow` instead of the components one.

edit: as I told you in DMs, the annotation `children` in `components/ui.ActionRow` isn't accurate imo, I suggest changing that to `AnyComponent` which is a Union of all the components, just a suggestion.